Source/WebKit/Platform/IPC/Connection.h

```diff
@@ -275,8 +275,38 @@ class Connection : public ThreadSafeRefCounted<Connection, WTF::DestructionThrea
 
     // Sync senders should check the SendSyncResult for true/false in case they need to know if the result was really received.
     // Sync senders should hold on to the SendSyncResult in case they reference the contents of the reply via DataRefererence / ArrayReference.
-    using SendSyncResult = std::unique_ptr<Decoder>;
-    template<typename T> SendSyncResult sendSync(T&& message, typename T::Reply&& reply, uint64_t destinationID, Timeout = Timeout::infinity(), OptionSet<SendSyncOption> sendSyncOptions = { }); // Main thread only.
+
+    using SendSyncLegacyResult = std::unique_ptr<Decoder>;
+    template<typename T> SendSyncLegacyResult sendSync(T&& message, typename T::Reply&& reply, uint64_t destinationID, Timeout = Timeout::infinity(), OptionSet<SendSyncOption> sendSyncOptions = { }); // Main thread only.
+
+    template<typename T>
+    struct SendSyncResult {
```

Comment:
You could still investigate if you can mark up the SendSyncResult in following way:


```
template<typename T> struct SendSyncResult : public T::ReplyArguments
{
     std::unique_ptr<Decoder> decoder;
     ...
 };
 
 std::optional<SendSyncResult<T>> Connection::sendSync(...);
 
....

void XXXX::GetIsSpeaking() 
{
     return sendSync(Messages::WebPageProxy::GetIsSpeaking()).valueOr(false);
}
```

Or some simpler variant of that.
So what I'd find simpler would be
std::optional<SendSyncResult<T>>
instead of
SendSyncResult<T> with bool operator and std::optional arguments.

Reply:
this suggestion is not mandatory, just something to consider now that you're up-to-speed with the search-and-replace.

Reply:
This trips up the `takeReplyOr(...)` mechanics, where the provided values can be used for initialization if the reply arguments aren't available. With this change, std::optional<> would first have to be validated. There's ways like standalone helper functions that could do this instead, but I don't think it would turn out easily readable.